### PR TITLE
409 Add `Idp Client Id` to `Application`

### LINKED ISSUE #409: [Portal Backend] Add IDP_Client_ID as an attribute to the Application
### Problem

The current definition for the `Application` not having `IdP Client ID` as an attribute. But It is essential to know that attribute, which need to be store in the database. Because decoding the `appToken` and finding the `IdP Client ID` is the only way to identify which application is accessing the data from the Orchestration Engine.

### Proposed Solution

- [x] Add `IdP Client ID` as an attribute to the `Application` model
- [x] During the `Create` operation for `Application` model. At first, system should make a request to the Identity Provider (IdP) for creating an `M2M` application in the `IdP` and we should store the `Client ID` return by this operation in our database
- [x] Implement a GET endpoint in the `portal-backend` that will return the `application_id` when we make request with the parameter `IdP-Client-ID`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by sthanikan2000**
## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made
- Add IdP Client ID as an attribute to the Application model
- During the Create operation for Application model. At first, system should make a request to the Identity Provider (IdP) for creating an M2M application in the IdP and we should store the Client ID return by this operation in our database
- Implement a GET endpoint in the portal-backend that will return the application_id when we make request with the parameter IdP-Client-ID

## Testing
- [x] I have tested this change locally
- [x] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

## Checklist
- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues
Closes #409

### CODE REVIEW

File: portal-backend/v1/services/application_service.go [hunk 1]
```diff
@@ -144,10 +191,25 @@ func (s *ApplicationService) GetApplication(applicationID string) (*models.Appli
 	return response, nil
 }
 
+// GetApplicationIdByIdpClientId retrives applicationId by idpClientId
+func (s *ApplicationService) GetApplicationIdByIdpClientId(ctx context.Context, idpClientId string) (*models.ApplicationIDResponse, error) {
+	var application models.Application
+	err := s.db.WithContext(ctx).First(&application, "idp_client_id = ?", idpClientId).Error
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
```
Comment (ginaxu1) [inline portal-backend/v1/services/application_service.go]: direct comparison not following Go error handling best practices. better:

```go
import "errors"

if errors.Is(err, gorm.ErrRecordNotFound) {
```

File: portal-backend/v1/services/application_service.go [hunk 2]
```diff
@@ -144,10 +191,25 @@ func (s *ApplicationService) GetApplication(applicationID string) (*models.Appli
 	return response, nil
 }
 
+// GetApplicationIdByIdpClientId retrives applicationId by idpClientId
+func (s *ApplicationService) GetApplicationIdByIdpClientId(ctx context.Context, idpClientId string) (*models.ApplicationIDResponse, error) {
```
Comment (ginaxu1) [inline portal-backend/v1/services/application_service.go]: missing `idpClientId` validation. suggest:

```go
func (s *ApplicationService) GetApplicationIdByIdpClientId(ctx context.Context, idpClientId string) (*string, error) {
	if idpClientId == "" {
		return nil, fmt.Errorf("idpClientId cannot be empty")
	}
	// ... rest of function
}
```

**Review by ginaxu1 — CHANGES_REQUESTED** [review]

File: portal-backend/v1/services/application_service.go [hunk 3]
```diff
@@ -14,40 +16,71 @@ import (
...
 		return nil, fmt.Errorf("failed to create application: %w", err)
 	}
+	appOIDCInfo, err := s.idp.GetApplicationOIDC(ctx, *idpApplicationID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get application OIDC: %w", err)
+	}
+
+	// Step 2: Create application in database
+	application := models.Application{
+		ApplicationID:          uuid.New().String(),
```
[the remark this answers is not in the document]
Comment (sthanikan2000) [reply portal-backend/v1/services/application_service.go]: This change is intentional
